static/js/millennium_diff_table.js

```diff
@@ -0,0 +1,574 @@
+        break;
+      case "search-type":
+        newResource.searchSupported = true;
+        break;
+      case "create":
+        newResource.createSupported = true;
+        break;
+      case "update":
+        newResource.updateSupported = true;
+        break;
+      case "patch":
+        newResource.patchSupported = true;
+        break;
+      case "delete":
+        newResource.deleteSupported = true;
+        break;
+    }
+  }
+
+  return newResource;
+}
+
+/**
+ * Creates an object of matching DSTU2 and R4 resources.
+ * @param {object} resource - A DSTU2 resource object.
+ * @param {array} resourceArray - The array of R4 resources objects.
+ * @returns {object} Contains arrays of DSTU2 and R4 resources that match.
+ * @returns {object} Contains an array of only DSTU2 resources if no R4 matches are found.
+ */
+function buildMatch(resource, resourceArray) {
+  let matchingResource = null;
+  for (let i = 0; i < resourceArray.length; i++) {
+    if (resourceArray[i] != null) {
+      if (resourceArray[i].type === resource.type) {
+        matchingResource = resourceArray[i];
+        delete resourceArray[i];
+      }
+    }
+  }
+  if (matchingResource != null) {
+    return {
+      dstu2Resources: [buildResource(resource)],
+      r4Resources: [buildResource(matchingResource)]
+    };
+  }
+  else {
+    return {
+      dstu2Resources: [buildResource(resource)]
+    };
+  }
+}
+
+/**
+ * Creates an object of matching DSTU2 and R4 resources that require special configuration.
+ * @param {object} config - The configuration object from resourceConfig.
+ * @param {array} dstu2Resources - The array of DSTU2 resource objects.
+ * @param {array} r4Resources - The array of R4 resource objects.
+ * @returns {object} Contains arrays of DSTU2 and R4 resources that match.
+ */
+function buildMatchFromConfig(config, dstu2Resources, r4Resources) {
+  let configuredResource = {
+    dstu2Resources: [],
+    r4Resources: [],
+  };
+
+  // Find the DSTU2 resources in the config
+  for (let i = 0; i < config.resourcesCombined.length; i++) {
+    for (let j = 0; j < dstu2Resources.length; j++) {
+      if (dstu2Resources[j] != null) {
+        if (dstu2Resources[j].type === config.resourcesCombined[i]) {
+          configuredResource.dstu2Resources.push(buildResource(dstu2Resources[j]));
+          delete dstu2Resources[j];
+        }
+      }
+    }
+  }
+
+  // Find the R4 resources in the config
+  for (let i = 0; i < config.combinedTo.length; i++) {
+    for (let j = 0; j < r4Resources.length; j++) {
+      if (r4Resources[j] != null) {
+        if (r4Resources[j].type === config.combinedTo[i]) {
+          configuredResource.r4Resources.push(buildResource(r4Resources[j]));
+          delete r4Resources[j];
+        }
+      }
+    }
+  }
+
+  if (config.hasOwnProperty('notes')) {
+    configuredResource.notes = config.notes;
+  }
+  return configuredResource;
+}
+
+/**
+ * Matches the resources between DSTU2 and R4.
```

Comment:
It might be best if we describe a little bit more _how_ we are matching resources between DSTU2 and R4 it'll make it clearer to future maintainers how this is done.

Reply:
Updated here: https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R237-R239
https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R173-R177
https://github.com/cerner/fhir.cerner.com/pull/416/commits/de755cb18506fe0370836bfed9a96d57296d7080#diff-4066a5d03340f019ca6531e5bc567919R204-R207